### ISSUE TITLE
Add "Introduce variable" feature

### ISSUE THREAD

**Issue by webxl** [hax/refactor#8]
This is particularly helpful when you want to assign part of a series chained methods into a reusable value.

Edit: Also support multiple replacements in the current block (if applicable)

**Comment by PhiLhoSoft** [comment]
Yes, that's what the https://atom.io/packages/extract-selection package does, but for _refactor_ to merit its name, it should increase the palette of refactorings... And having all refactorings within the same package would be nice.

Another possible refactoring: allows to reorder the list of parameters of a function. Like in Eclipse. Difficulty: also change the call sites... Harder in JS than in Java!

**Comment by webxl** [comment]
That plugin only partially does what I'm looking for. I'm just spoiled by Resharper and Webstorm, I guess. Their implementation could optionally replace all occurrences of the expression in the block and place the assignment at the appropriate spot, e.g. above the first use of the expression in the function in the case of JS.
